build-logic/src/main/kotlin/dokkabuild/testing/android-setup.gradle.kts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package dokkabuild.testing
+
+/**
+ * Utilities for preparing Android projects for Integration Tests
+ */
+
+plugins {
+    base
+}
+
+val localAndroidSdkRoot: DirectoryProperty = objects.directoryProperty().apply {
+    convention(layout.projectDirectory.dir("projects/ANDROID_SDK"))
+}
+
+/**
+ * Create an `local.properties` file so that Android projects will automatically work in integration tests.
+ */
+abstract class CreateAndroidLocalPropertiesFile : DefaultTask() {
+    @get:OutputFile
+    abstract val localPropertiesFile: RegularFileProperty
+
+    /**
+     * Location of the Android SDK.
+     * Will be automatically determined by the environment variable `ANDROID_SDK_ROOT`, or
+     */
+    @get:Input
+    // use a string not a dir because we don't actually care about the contents of the directory
+    abstract val androidSdkDirPath: Property<String>
+
+    @TaskAction
+    fun action() {
+        val taskPath = path
+
+        localPropertiesFile.asFile.get().apply {
+            parentFile.mkdirs()
+            createNewFile()
+            writeText(
+                """
+                  |# DO NOT EDIT - Generated by $taskPath
+                  |
+                  |sdk.dir=${androidSdkDirPath.get()}
+                  |
+                """.trimMargin()
+            )
+        }
+    }
+}
+
+val createAndroidLocalPropertiesFile by tasks.registering(CreateAndroidLocalPropertiesFile::class) {
+    localPropertiesFile.convention(layout.file(provider { temporaryDir.resolve("local.properties") }))
+    androidSdkDirPath.convention(providers
+        // first try getting the SDK installed on via GitHub step setup-android
```

Comment:
Is it possible to somehow easily add out-of-the-box support for reading this value from `local.properties` inside project? As I already have Android SDK installed at my side.

Reply:
That's a good idea, I'd be happy to do it. [I've done something similar in MockK before](https://github.com/mockk/mockk/blob/master/buildSrc/android-sdk-detector.settings.gradle.kts). However I'm a little bit reluctant because the Android SDK is only needed in integration tests, and scanning/parsing `local.properties` files would add some complexity & maintenance. 

Since it's possible to set the Android SDK using an environment variable, I'd prefer not to. But what do you think?